substrate/frame/revive/rpc/src/client.rs

```diff
@@ -342,8 +398,23 @@ impl Client {
 		let last = self.latest_block().await.number().saturating_sub(1);
 		let range = last.saturating_sub(index_last_n_blocks)..last;
 		log::info!(target: LOG_TARGET, "🗄️ Indexing past blocks in range {range:?}");
+
 		self.subscribe_past_blocks(range, |block| async move {
-			self.receipt_provider.insert_block_receipts(&block).await?;
+			// Special handling for genesis block (block 0)
+			// Genesis block is never stored on-chain, so we need to build it manually
+			let ethereum_hash = if block.number() == 0 {
+				let evm_block = self
+					.build_genesis_block(&block)
+					.await
+					.ok_or(ClientError::EthereumBlockNotFound)?;
+				evm_block.hash
```

Comment:
I wonder if we should not instead add some code to the genesisconfig to build this genesisblock if needed (I guess that would mostly be some config used by dev-node)

Reply:
Done as advised.
https://github.com/paritytech/polkadot-sdk/commit/9449a6d8daf71e2290bdb94e7e6f483270fa17c7